### DISCUSSION TITLE
Instrucciones para Configurar el Entorno de Desarrollo Usando Docker Compose

### DISCUSSION THREAD

**Discussion by JuanCarlosRL** [Aparking/AparKing_Backend#100] [Announcements]
Para simplificar el proceso de desarrollo y asegurar que todos los miembros del equipo puedan trabajar en un entorno consistente sin necesidad de configuraciones adicionales, se ha configurado Docker Compose 🚀. **A continuación se definen los pasos para configurar el entorno de desarrollo:**

> **Nota:** Este proceso puede tardar más de 10 minutos. Paciencia 🙂.

### Pasos para la Configuración 🛠️

1. **Instalación de Docker:**
   - Descarga e instala Docker desde [el sitio oficial de Docker](https://www.docker.com/products/docker-desktop) 💾 y termina el proceso de configuración de la aplicación de Docker antes de continuar.

2. **Construcción del Proyecto:**
   - Abre un terminal en la raíz del proyecto 📂.
   - Ejecuta el comando `docker-compose build` para construir los servicios definidos en el archivo `docker-compose.yml` 🏗️.

3. **Ejecución del Proyecto:**
   - Una vez completada la construcción, ejecuta `docker-compose up -d` para iniciar los contenedores en modo detached 🚀.

4. **Detener los Contenedores:**
   - Para detener y eliminar los contenedores, utiliza el comando `docker-compose down` 🛑.

5. **Acceso al Backend:**
   - La ruta para acceder al backend del proyecto es `localhost:3000` 🔗

### Comandos Adicionales 📌

- **Eliminar y Detener Volumen:**
   - Si necesitas detener los contenedores y eliminar los volúmenes asociados, utiliza `docker-compose down -v` 🗑️. Ten en cuenta que necesitarás realizar el proceso de construcción (`docker-compose build`) nuevamente después de este paso.

> **Notas Importantes:**
> - Si agregas nuevas dependencias al proyecto, será necesario realizar nuevamente el `docker-compose build` 🔧.
> - Si encuentras problemas al intentar ejecutar los contenedores, es posible que los puertos requeridos estén siendo utilizados por otro proceso ⚠️.

**Comment by JoseJavierAlcobendas** [comment]
Al acabr de instalar el docker y ejecutar el Docker Desktop me sale este error 
![image](https://github.com/Aparking/AparKing_Backend/assets/73230697/dfb63cbe-a4f1-42cf-a7fb-cd9e062f700f)
@JuanCarlosRL

**Comment by PedroRuizAguilar** [comment]
@JuanCarlosRL A mi tb me ha dado un error al hacer el comando de docker-compose build 
![image](https://github.com/Aparking/AparKing_Backend/assets/72878120/b23f3894-3330-41aa-80cd-af7289bc968a)
